IMPORTING THE DATA

In [1]:
import pandas as pd
df=pd.read_csv("Automobile_insurance_fraud.csv")
print (df)

     months_as_customer  age  policy_number policy_bind_date policy_state  \
0                   328   48         521585       17-10-2014           OH   
1                   228   42         342868       27-06-2006           IN   
2                   134   29         687698       06-09-2000           OH   
3                   256   41         227811       25-05-1990           IL   
4                   228   44         367455       06-06-2014           IL   
..                  ...  ...            ...              ...          ...   
995                   3   38         941851       16-07-1991           OH   
996                 285   41         186934       05-01-2014           IL   
997                 130   34         918516       17-02-2003           OH   
998                 458   62         533940       18-11-2011           IL   
999                 456   60         556080       11-11-1996           OH   

    policy_csl  policy_deductable  policy_annual_premium  umbrella_limit  \

In [2]:
df.shape

(1000, 40)

In [3]:
df.head(10)

months_as_customer  age  policy_number policy_bind_date policy_state  \
0                 328   48         521585       17-10-2014           OH   
1                 228   42         342868       27-06-2006           IN   
2                 134   29         687698       06-09-2000           OH   
3                 256   41         227811       25-05-1990           IL   
4                 228   44         367455       06-06-2014           IL   
5                 256   39         104594       12-10-2006           OH   
6                 137   34         413978       04-06-2000           IN   
7                 165   37         429027       03-02-1990           IL   
8                  27   33         485665       05-02-1997           IL   
9                 212   42         636550       25-07-2011           IL   

  policy_csl  policy_deductable  policy_annual_premium  umbrella_limit  \
0    250/500               1000                1406.91               0   
1    250/500               2000                1197.22         5000000   
2    100/300               2000                1413.14         5000000   
3    250/500               2000                1415.74         6000000   
4   500/1000               1000                1583.91         6000000   
5    250/500               1000                1351.10               0   
6    250/500               1000                1333.35               0   
7    100/300               1000                1137.03               0   
8    100/300                500                1442.99               0   
9    100/300                500                1315.68               0   

   insured_zip  ... police_report_available total_claim_amount injury_claim  \
0       466132  ...                     YES              71610         6510   
1       468176  ...                       ?               5070          780   
2       430632  ...                      NO              34650         7700   
3       608117  ...                      NO              63400         6340   
4       610706  ...                      NO               6500         1300   
5       478456  ...                      NO              64100         6410   
6       441716  ...                       ?              78650        21450   
7       603195  ...                     YES              51590         9380   
8       601734  ...                     YES              27700         2770   
9       600983  ...                       ?              42300         4700   

  property_claim vehicle_claim  auto_make  auto_model auto_year  \
0          13020         52080       Saab         92x      2004   
1            780          3510   Mercedes        E400      2007   
2           3850         23100      Dodge         RAM      2007   
3           6340         50720  Chevrolet       Tahoe      2014   
4            650          4550     Accura         RSX      2009   
5           6410         51280       Saab          95      2003   
6           7150         50050     Nissan  Pathfinder      2012   
7           9380         32830       Audi          A5      2015   
8           2770         22160     Toyota       Camry      2012   
9           4700         32900       Saab         92x      1996   

  fraud_reported _c39  
0              Y  NaN  
1              Y  NaN  
2              N  NaN  
3              Y  NaN  
4              N  NaN  
5              Y  NaN  
6              N  NaN  
7              N  NaN  
8              N  NaN  
9              N  NaN  

[10 rows x 40 columns]

In [4]:
df.dtypes

months_as_customer               int64
age                              int64
policy_number                    int64
policy_bind_date                object
policy_state                    object
policy_csl                      object
policy_deductable                int64
policy_annual_premium          float64
umbrella_limit                   int64
insured_zip                      int64
insured_sex                     object
insured_education_level         object
insured_occupation              object
insured_hobbies                 object
insured_relationship            object
capital-gains                    int64
capital-loss                     int64
incident_date                   object
incident_type                   object
collision_type                  object
incident_severity               object
authorities_contacted           object
incident_state                  object
incident_city                   object
incident_location               object
incident_hour_of_the_day 

FEATURE ENGINEERING

In [6]:
df['Date']=df['policy_bind_date'].str.split('-').str[0]
df['Month']=df['policy_bind_date'].str.split('-').str[1]
df['Year']=df['policy_bind_date'].str.split('-').str[2]

In [7]:
df['Date']=df['Date'].astype(int)
df['Month']=df['Month'].astype(int)
df['Year']=df['Year'].astype(int)

In [8]:
df=df.drop(['policy_bind_date'],axis=1)

In [9]:
df.head()

months_as_customer  age  policy_number policy_state policy_csl  \
0                 328   48         521585           OH    250/500   
1                 228   42         342868           IN    250/500   
2                 134   29         687698           OH    100/300   
3                 256   41         227811           IL    250/500   
4                 228   44         367455           IL   500/1000   

   policy_deductable  policy_annual_premium  umbrella_limit  insured_zip  \
0               1000                1406.91               0       466132   
1               2000                1197.22         5000000       468176   
2               2000                1413.14         5000000       430632   
3               2000                1415.74         6000000       608117   
4               1000                1583.91         6000000       610706   

  insured_sex  ... property_claim vehicle_claim  auto_make auto_model  \
0        MALE  ...          13020         52080       Saab        92x   
1        MALE  ...            780          3510   Mercedes       E400   
2      FEMALE  ...           3850         23100      Dodge        RAM   
3      FEMALE  ...           6340         50720  Chevrolet      Tahoe   
4        MALE  ...            650          4550     Accura        RSX   

   auto_year  fraud_reported _c39 Date Month  Year  
0       2004               Y  NaN   17    10  2014  
1       2007               Y  NaN   27     6  2006  
2       2007               N  NaN    6     9  2000  
3       2014               Y  NaN   25     5  1990  
4       2009               N  NaN    6     6  2014  

[5 rows x 42 columns]

In [10]:
df['policy']=df['policy_csl'].str.split('/').str[0]
df['csl']=df['policy_csl'].str.split('/').str[1]

In [11]:
df.head()

months_as_customer  age  policy_number policy_state policy_csl  \
0                 328   48         521585           OH    250/500   
1                 228   42         342868           IN    250/500   
2                 134   29         687698           OH    100/300   
3                 256   41         227811           IL    250/500   
4                 228   44         367455           IL   500/1000   

   policy_deductable  policy_annual_premium  umbrella_limit  insured_zip  \
0               1000                1406.91               0       466132   
1               2000                1197.22         5000000       468176   
2               2000                1413.14         5000000       430632   
3               2000                1415.74         6000000       608117   
4               1000                1583.91         6000000       610706   

  insured_sex  ...  auto_make auto_model auto_year fraud_reported  _c39  Date  \
0        MALE  ...       Saab        92x      2004              Y   NaN    17   
1        MALE  ...   Mercedes       E400      2007              Y   NaN    27   
2      FEMALE  ...      Dodge        RAM      2007              N   NaN     6   
3      FEMALE  ...  Chevrolet      Tahoe      2014              Y   NaN    25   
4        MALE  ...     Accura        RSX      2009              N   NaN     6   

  Month  Year policy   csl  
0    10  2014    250   500  
1     6  2006    250   500  
2     9  2000    100   300  
3     5  1990    250   500  
4     6  2014    500  1000  

[5 rows x 44 columns]

In [12]:
df['policy']=df['policy'].astype(int)
df['csl']=df['csl'].astype(int)

In [13]:
df=df.drop(['policy_csl'],axis=1)
df=df.drop(['auto_model'],axis=1)

In [14]:
df=df.drop(['_c39'],axis=1)

In [15]:
df.head()

months_as_customer  age  policy_number policy_state  policy_deductable  \
0                 328   48         521585           OH               1000   
1                 228   42         342868           IN               2000   
2                 134   29         687698           OH               2000   
3                 256   41         227811           IL               2000   
4                 228   44         367455           IL               1000   

   policy_annual_premium  umbrella_limit  insured_zip insured_sex  \
0                1406.91               0       466132        MALE   
1                1197.22         5000000       468176        MALE   
2                1413.14         5000000       430632      FEMALE   
3                1415.74         6000000       608117      FEMALE   
4                1583.91         6000000       610706        MALE   

  insured_education_level  ... property_claim vehicle_claim  auto_make  \
0                      MD  ...          13020         52080       Saab   
1                      MD  ...            780          3510   Mercedes   
2                     PhD  ...           3850         23100      Dodge   
3                     PhD  ...           6340         50720  Chevrolet   
4               Associate  ...            650          4550     Accura   

   auto_year  fraud_reported Date Month  Year policy   csl  
0       2004               Y   17    10  2014    250   500  
1       2007               Y   27     6  2006    250   500  
2       2007               N    6     9  2000    100   300  
3       2014               Y   25     5  1990    250   500  
4       2009               N    6     6  2014    500  1000  

[5 rows x 41 columns]

In [16]:
df['insured_occupation'].value_counts()

machine-op-inspct    93
prof-specialty       85
tech-support         78
sales                76
exec-managerial      76
craft-repair         74
transport-moving     72
other-service        71
priv-house-serv      71
armed-forces         69
adm-clerical         65
protective-serv      63
handlers-cleaners    54
farming-fishing      53
Name: insured_occupation, dtype: int64

In [17]:
df['insured_hobbies'].value_counts()

reading           64
paintball         57
exercise          57
bungie-jumping    56
camping           55
movies            55
golf              55
kayaking          54
yachting          53
hiking            52
video-games       50
base-jumping      49
skydiving         49
board-games       48
polo              47
chess             46
dancing           43
sleeping          41
cross-fit         35
basketball        34
Name: insured_hobbies, dtype: int64

In [18]:
df['incident_date'].value_counts()

02-02-2015    28
17-02-2015    26
07-01-2015    25
10-01-2015    24
24-01-2015    24
04-02-2015    24
19-01-2015    23
08-01-2015    22
30-01-2015    21
13-01-2015    21
12-02-2015    20
31-01-2015    20
22-02-2015    20
06-02-2015    20
01-01-2015    19
21-01-2015    19
14-01-2015    19
21-02-2015    19
23-02-2015    19
12-01-2015    19
28-02-2015    18
03-01-2015    18
14-02-2015    18
01-02-2015    18
18-01-2015    18
20-01-2015    18
25-02-2015    18
09-01-2015    17
24-02-2015    17
26-02-2015    17
06-01-2015    17
08-02-2015    17
16-01-2015    16
16-02-2015    16
05-02-2015    16
13-02-2015    16
15-02-2015    16
17-01-2015    15
28-01-2015    15
15-01-2015    15
18-02-2015    15
22-01-2015    14
27-02-2015    14
20-02-2015    14
09-02-2015    13
03-02-2015    13
23-01-2015    13
27-01-2015    13
04-01-2015    12
01-03-2015    12
29-01-2015    11
26-01-2015    11
02-01-2015    11
10-02-2015    10
19-02-2015    10
11-02-2015    10
25-01-2015    10
07-02-2015    10
11-01-2015    

In [19]:
df['collision_type'].value_counts()

Rear Collision     292
Side Collision     276
Front Collision    254
?                  178
Name: collision_type, dtype: int64

In [20]:
df['incident_severity'].value_counts()

Minor Damage      354
Total Loss        280
Major Damage      276
Trivial Damage     90
Name: incident_severity, dtype: int64

In [21]:
df['police_report_available'].value_counts()

?      343
NO     343
YES    314
Name: police_report_available, dtype: int64

In [22]:
df['incident_state'].value_counts()

NY    262
SC    248
WV    217
VA    110
NC    110
PA     30
OH     23
Name: incident_state, dtype: int64

In [23]:
df['incident_city'].value_counts()

Springfield    157
Arlington      152
Columbus       149
Northbend      145
Hillsdale      141
Riverwood      134
Northbrook     122
Name: incident_city, dtype: int64

In [24]:
df['incident_location'].value_counts()

1325 1st Lane          1
6044 Weaver Drive      1
9317 Apache Ave        1
6012 Texas Hwy         1
8477 Francis Hwy       1
                      ..
3087 Oak Hwy           1
2003 Maple Hwy         1
4055 2nd Drive         1
3790 Andromedia Hwy    1
4296 Pine Hwy          1
Name: incident_location, Length: 1000, dtype: int64

In [25]:
df['property_damage'].value_counts()

?      360
NO     338
YES    302
Name: property_damage, dtype: int64

In [26]:
df['policy'].value_counts()

250    351
100    349
500    300
Name: policy, dtype: int64

In [27]:
df=df.drop(['insured_occupation'],axis=1)
df=df.drop(['insured_hobbies'],axis=1)
df=df.drop(['insured_relationship'],axis=1)
df=df.drop(['incident_location'],axis=1)

In [28]:
df['Date1']=df['incident_date'].str.split('-').str[0]
df['Month1']=df['incident_date'].str.split('-').str[1]
df['Year1']=df['incident_date'].str.split('-').str[2]

In [29]:
df['Date1']=df['Date1'].astype(int)
df['Month1']=df['Month1'].astype(int)
df['Year1']=df['Year1'].astype(int)

In [30]:
df['collision_type']=df['collision_type'].str.replace('?','')

In [31]:
df['collision_type'].value_counts()

Rear Collision     292
Side Collision     276
Front Collision    254
                   178
Name: collision_type, dtype: int64

In [32]:
df['collision_type'].fillna(df['collision_type'].value_counts().index[0],inplace=True)

In [33]:
df['collision_type'].isnull().sum()

0

In [34]:
df['property_damage']=df['property_damage'].str.replace('?','')

In [35]:
df['property_damage'].value_counts()

       360
NO     338
YES    302
Name: property_damage, dtype: int64

In [36]:
df['property_damage'].fillna(df['property_damage'].value_counts().index[0],inplace=True)

In [37]:
df['property_damage'].isnull().sum()

0

In [38]:
df['incident']=df['incident_severity'].str.split('').str[0]
df['severity']=df['incident_severity'].str.split('').str[1]

In [39]:
df['collision']=df['collision_type'].str.split('').str[0]
df['type']=df['collision_type'].str.split('').str[1]

In [40]:
df.head()

months_as_customer  age  policy_number policy_state  policy_deductable  \
0                 328   48         521585           OH               1000   
1                 228   42         342868           IN               2000   
2                 134   29         687698           OH               2000   
3                 256   41         227811           IL               2000   
4                 228   44         367455           IL               1000   

   policy_annual_premium  umbrella_limit  insured_zip insured_sex  \
0                1406.91               0       466132        MALE   
1                1197.22         5000000       468176        MALE   
2                1413.14         5000000       430632      FEMALE   
3                1415.74         6000000       608117      FEMALE   
4                1583.91         6000000       610706        MALE   

  insured_education_level  ...  Year  policy   csl Date1 Month1 Year1  \
0                      MD  ...  2014     250   500    25      1  2015   
1                      MD  ...  2006     250   500    21      1  2015   
2                     PhD  ...  2000     100   300    22      2  2015   
3                     PhD  ...  1990     250   500    10      1  2015   
4               Associate  ...  2014     500  1000    17      2  2015   

  incident severity collision  type  
0                 M               S  
1                 M                  
2                 M               R  
3                 M               F  
4                 M                  

[5 rows x 44 columns]

In [41]:
df=df.drop(['incident_type'],axis=1)
df=df.drop(['collision_type'],axis=1)

In [42]:
df['police_report_available']=df['police_report_available'].str.replace('?','')

In [43]:
df['police_report_available'].value_counts()

NO     343
       343
YES    314
Name: police_report_available, dtype: int64

In [44]:
df['police_report_available'].fillna(df['police_report_available'].value_counts().index[0],inplace=True)

In [45]:
df['police_report_available'].isnull().sum()

0

In [46]:
df=df.drop(['insured_sex'],axis=1)
df=df.drop(['incident_date'],axis=1)

ENCODING

In [47]:
from sklearn.preprocessing import LabelEncoder
LE=LabelEncoder()
df["policy_state"]=LE.fit_transform(df['policy_state'])
df["auto_make"]=LE.fit_transform(df['auto_make'])
df["fraud_reported"]=LE.fit_transform(df['fraud_reported'])
df["insured_education_level"]=LE.fit_transform(df['insured_education_level'])
df["fraud_reported"]=LE.fit_transform(df['fraud_reported'])
df["insured_education_level"]=LE.fit_transform(df['insured_education_level'])
df["authorities_contacted"]=LE.fit_transform(df['authorities_contacted'])
df["incident_state"]=LE.fit_transform(df['incident_state'])
df["incident_city"]=LE.fit_transform(df['incident_city'])
df["property_damage"]=LE.fit_transform(df['property_damage'])
df["police_report_available"]=LE.fit_transform(df['police_report_available'])
df["incident"]=LE.fit_transform(df['incident'])
df["severity"]=LE.fit_transform(df['severity'])
df["collision"]=LE.fit_transform(df['collision'])
df["type"]=LE.fit_transform(df['type'])

In [48]:
df["incident_severity"]=LE.fit_transform(df['incident_severity'])

In [49]:
df.dtypes

months_as_customer               int64
age                              int64
policy_number                    int64
policy_state                     int32
policy_deductable                int64
policy_annual_premium          float64
umbrella_limit                   int64
insured_zip                      int64
insured_education_level          int64
capital-gains                    int64
capital-loss                     int64
incident_severity                int32
authorities_contacted            int32
incident_state                   int32
incident_city                    int32
incident_hour_of_the_day         int64
number_of_vehicles_involved      int64
property_damage                  int32
bodily_injuries                  int64
witnesses                        int64
police_report_available          int32
total_claim_amount               int64
injury_claim                     int64
property_claim                   int64
vehicle_claim                    int64
auto_make                

In [50]:
df.head()

months_as_customer  age  policy_number  policy_state  policy_deductable  \
0                 328   48         521585             2               1000   
1                 228   42         342868             1               2000   
2                 134   29         687698             2               2000   
3                 256   41         227811             0               2000   
4                 228   44         367455             0               1000   

   policy_annual_premium  umbrella_limit  insured_zip  \
0                1406.91               0       466132   
1                1197.22         5000000       468176   
2                1413.14         5000000       430632   
3                1415.74         6000000       608117   
4                1583.91         6000000       610706   

   insured_education_level  capital-gains  ...  Year  policy   csl  Date1  \
0                        4          53300  ...  2014     250   500     25   
1                        4              0  ...  2006     250   500     21   
2                        6          35100  ...  2000     100   300     22   
3                        6          48900  ...  1990     250   500     10   
4                        0          66000  ...  2014     500  1000     17   

   Month1  Year1  incident  severity  collision  type  
0       1   2015         0         0          0     3  
1       1   2015         0         0          0     0  
2       2   2015         0         0          0     2  
3       1   2015         0         0          0     1  
4       2   2015         0         0          0     0  

[5 rows x 40 columns]

In [51]:
import numpy as np
df.isnull().sum()

months_as_customer             0
age                            0
policy_number                  0
policy_state                   0
policy_deductable              0
policy_annual_premium          0
umbrella_limit                 0
insured_zip                    0
insured_education_level        0
capital-gains                  0
capital-loss                   0
incident_severity              0
authorities_contacted          0
incident_state                 0
incident_city                  0
incident_hour_of_the_day       0
number_of_vehicles_involved    0
property_damage                0
bodily_injuries                0
witnesses                      0
police_report_available        0
total_claim_amount             0
injury_claim                   0
property_claim                 0
vehicle_claim                  0
auto_make                      0
auto_year                      0
fraud_reported                 0
Date                           0
Month                          0
Year      

SPLITTING INTO X AND Y

In [52]:
x=df.drop("fraud_reported",axis=1)
y=df["fraud_reported"]
x.head()

months_as_customer  age  policy_number  policy_state  policy_deductable  \
0                 328   48         521585             2               1000   
1                 228   42         342868             1               2000   
2                 134   29         687698             2               2000   
3                 256   41         227811             0               2000   
4                 228   44         367455             0               1000   

   policy_annual_premium  umbrella_limit  insured_zip  \
0                1406.91               0       466132   
1                1197.22         5000000       468176   
2                1413.14         5000000       430632   
3                1415.74         6000000       608117   
4                1583.91         6000000       610706   

   insured_education_level  capital-gains  ...  Year  policy   csl  Date1  \
0                        4          53300  ...  2014     250   500     25   
1                        4              0  ...  2006     250   500     21   
2                        6          35100  ...  2000     100   300     22   
3                        6          48900  ...  1990     250   500     10   
4                        0          66000  ...  2014     500  1000     17   

   Month1  Year1  incident  severity  collision  type  
0       1   2015         0         0          0     3  
1       1   2015         0         0          0     0  
2       2   2015         0         0          0     2  
3       1   2015         0         0          0     1  
4       2   2015         0         0          0     0  

[5 rows x 39 columns]

In [53]:
y.head()

0    1
1    1
2    0
3    1
4    0
Name: fraud_reported, dtype: int64

TRAIN TEST SPLIT

In [54]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.33, random_state=0) 

BUILDING MODELS

RANDOM FOREST

In [55]:
from sklearn.ensemble import RandomForestClassifier
classifier = RandomForestClassifier(n_estimators = 10, criterion = 'entropy', random_state = 0)
classifier.fit(x_train, y_train)

RandomForestClassifier(criterion='entropy', n_estimators=10, random_state=0)

In [56]:
y_pred = classifier.predict(x_test)
y_pred

array([0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0,
       1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,

In [57]:
from sklearn import metrics
print('The accuracy of Random Forest Classification is: ', metrics.accuracy_score(y_pred, y_test))

The accuracy of Random Forest Classification is:  0.7848484848484848


In [58]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)

In [59]:
cm 

array([[227,  21],
       [ 50,  32]], dtype=int64)

DTC

In [60]:
from sklearn import tree
from sklearn.tree import DecisionTreeClassifier
classifier = DecisionTreeClassifier(criterion = 'entropy', random_state = 0)
classifier.fit(x_train, y_train)

DecisionTreeClassifier(criterion='entropy', random_state=0)

In [61]:
y_pred = classifier.predict(x_test)
y_pred

array([0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0,
       1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0,
       1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1,
       0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0,
       0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0,
       1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0,
       0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0,

In [62]:
from sklearn import metrics
print('The accuracy of Random Forest Classification is: ', metrics.accuracy_score(y_pred, y_test))

The accuracy of Random Forest Classification is:  0.7272727272727273


In [63]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)

In [64]:
cm 

array([[201,  47],
       [ 43,  39]], dtype=int64)

Random forest is the best model with 78% accuracy